services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/attachment/application/ApplicationAttachmentMigratorITest.kt

```diff
@@ -0,0 +1,146 @@
+    @Autowired private val contentRepository: ApplicationAttachmentContentRepository,
+    @Autowired private val attachmentFactory: ApplicationAttachmentFactory,
+    @Autowired private val fileClient: MockFileClient,
+) : DatabaseTest() {
+
+    @BeforeEach
+    fun setup() {
+        clearAllMocks()
+        fileClient.recreateContainers()
+    }
+
+    @AfterEach
+    fun tearDown() {
+        checkUnnecessaryStub()
+    }
+
+    @Nested
+    @Transactional
+    inner class FindAttachmentWithDatabaseContent {
+        @Test
+        fun `Should return null if no attachments in content table`() {
+            val attachmentWithContent = attachmentMigrator.findAttachmentWithDatabaseContent()
+
+            assertThat(attachmentWithContent).isNull()
+        }
+
+        @Test
+        fun `Should return un-migrated content when there are any`() {
+            val attachment = attachmentFactory.save().withDbContent().value
+            assertThat(attachmentRepository.findAll()).hasSize(1)
+            assertThat(contentRepository.findAll()).hasSize(1)
+            val content = contentRepository.findByIdOrNull(attachment.id!!)!!
+            assertThat(fileClient.listBlobs(Container.HAKEMUS_LIITTEET)).isEmpty()
+
+            val attachmentWithContent = attachmentMigrator.findAttachmentWithDatabaseContent()
+
+            assertThat(attachmentWithContent).isNotNull().all {
+                prop(ApplicationAttachmentWithContent::id).isEqualTo(attachment.id!!)
+                prop(ApplicationAttachmentWithContent::applicationId)
+                    .isEqualTo(attachment.applicationId)
+                prop(ApplicationAttachmentWithContent::content).all {
+                    prop(AttachmentContent::fileName).isEqualTo(attachment.fileName)
+                    prop(AttachmentContent::contentType).isEqualTo(attachment.contentType)
+                    prop(AttachmentContent::bytes).isEqualTo(content.content)
+                }
+            }
+        }
+    }
+
+    @Nested
+    inner class Migrate {
+        @Test
+        fun `Should migrate content to blob`() {
+            attachmentFactory.save().withDbContent().value
+            assertThat(attachmentRepository.findAll()).hasSize(1)
+            assertThat(contentRepository.findAll()).hasSize(1)
+            assertThat(fileClient.listBlobs(Container.HAKEMUS_LIITTEET)).isEmpty()
+            val attachmentWithContent = attachmentMigrator.findAttachmentWithDatabaseContent()!!
+
+            val blobPath = attachmentMigrator.migrate(attachmentWithContent)
+
+            assertThat(attachmentRepository.findAll()).hasSize(1)
+            assertThat(contentRepository.findAll()).hasSize(1)
+            assertThat(fileClient.listBlobs(Container.HAKEMUS_LIITTEET)).hasSize(1)
+            assertThat(fileClient.listBlobs(Container.HAKEMUS_LIITTEET).first()).all {
+                prop(TestFile::path).isEqualTo(blobPath)
+            }
```

Comment:
`all {}` isn't necessary with just one assertation.
```suggestion
            assertThat(fileClient.listBlobs(Container.HAKEMUS_LIITTEET).first())
                .prop(TestFile::path)
                .isEqualTo(blobPath)
```

Reply:
Fixed